lib/readyset/query_annotator.rb

```diff
@@ -0,0 +1,18 @@
+require 'active_support/concern'
+
+module Readyset
+  module QueryAnnotator
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def route(prevent_writes: true, &block)
+        if Readyset.configuration.query_annotations
+          ActiveSupport::ExecutionContext.set(readyset_query: 'routed_to_readyset')
```

Comment:
Hmm `ActiveSupport::ExecutionContext` is an [undocumented class](https://github.com/rails/rails/blob/6b93fff8af32ef5e91f4ec3cfffb081d0553faf0/activesupport/lib/active_support/execution_context.rb#L4) in Rails, so we probably shouldn't use it here -- it could be changed/removed with minor Rails version updates.

I think we can integrate the way that the `ActiveRecord::QueryLogs` documentation suggests: api.rubyonrails.org/classes/ActiveSupport/QueryLogs.html
> New comment tags can be defined by adding them in a Hash to the tags Array. Tags can have dynamic content by setting a Proc or lambda value in the Hash, and can reference any value stored by Rails in the context object. [ActiveSupport::CurrentAttributes](https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html) can be used to store application values. Tags with nil values are omitted from the query comment.

Basically, we could do something like this to pass info about whether we're routing to ReadySet:
```ruby
Rails.configuration.active_record.query_log_tags << {
  routed_to_readyset?: ->(context) do
    Current.routing_to_readyset?
  end,
}
```
Where `Current` is defined like so:
```ruby
module Readyset
  class Current < ActiveSupport::CurrentAttributes
    attr_writer :routing_to_readyset

    def routing_to_readyset?
      !!@routing_to_readyset
    end
  end
end
```
Then we'd just need to update the `routing_to_readyset` attribute whenever `Readyset.route` is called:
```ruby
  def self.route(prevent_writes: true, &block)
    Current.routing_to_readyset = true
    ...
  ensure
    Current.routing_to_readyset = false
  end
```

Reply:
Hmm. I'm still figuring out how to check that the SQL comment was added. 

So far, it's been a struggle trying to get it to output the SQL query log with comments enabled. It might be a limitation with Combustion.

Reply:
I'd say we should probably switch to integrating via `ActiveRecord::QueryLogs` before spending too much time debugging the specs. It's possible there's something about using `ExecutionContext` that's causing the spec issue

Reply:
Yesterday it wasn't printing `/*application:Combustion*/` in `test.log` queries, but it is today! 

![image](https://github.com/readysettech/readyset-rails/assets/22648572/be5419d4-d7aa-4beb-905f-95f91b38f62f)


(╯°□°)╯︵ ┻━┻